tms/src/main/java/com/netcracker/edu/tms/dao/ProjectDaoJpaImpl.java

```diff
@@ -0,0 +1,140 @@
+        LOGGER.debug("getProjectById called with {}", projectId);
+        return entityManager.find(Project.class, projectId);
+    }
+
+    @Override
+    public Project getProjectByName(String name) {
+        Query query = entityManager.createQuery(QueryConsts.SELECT_WITH_NAME);
+        query.setParameter("name", name);
+        List<Project> ret = query.getResultList();
+
+        LOGGER.info("getProjectByName called with {}", name);
+        if (ret.isEmpty()) {
+            return null;
+        }
+        return ret.get(0);
+    }
+
+    @Override
+    public boolean addProject(Project project) {
+        entityManager.persist(project);
+        LOGGER.debug("addProject called with {}", project.toString());
+        return true;
+    }
+
+    @Override
+    public boolean updateProject(Project project, BigInteger updatedProjectId) {
+        if (entityManager.find(Project.class, updatedProjectId) != null) {
+            entityManager.merge(project);
+            LOGGER.debug("updateProject called with {}", project.toString());
+            return true;
+        } else {
+            LOGGER.warn("No such Project with ID= {} in updateProject appeared", updatedProjectId);
+            return false;
+        }
+    }
+
+    @Override
+    public boolean deleteProject(BigInteger projectId) {
+        Project same = entityManager.find(Project.class, projectId);
+
+        if (same != null) {
+            entityManager.remove(same);
+            LOGGER.debug("deleteProject called with {}", projectId.toString());
+            return true;
+        } else {
+            LOGGER.warn("No such Project with id={} in deleteProject appeared", projectId);
+            return false;
+        }
+    }
+
+    @Override
+    public List<Project> findProjectsByCreatorId(BigInteger creatorId) {
+        if (BigInteger.ZERO.compareTo(creatorId) >= 0) {
+            return Collections.emptyList();
+        }
+        Query query = entityManager.createQuery(QueryConsts.SELECT_PROJECTS_BY_CREATOR_ID);
+        query.setParameter("creator_id", creatorId);
+        List<Project> ret = query.getResultList();
+        LOGGER.debug("findProjectsByCreatorId called with {}", creatorId);
+        return ret;
+    }
+
+    @Override
+    public List<Project> getAllProjects() {
+        Query query = entityManager.createQuery(QueryConsts.SELECT_ALL);
+        List<Project> ret = query.getResultList();
+        LOGGER.debug("getAllProjects called");
+        return ret;
+    }
+
+    @Override
+    public List<Project> getUsersProjects(BigInteger userId) {
+        Query query = entityManager.createQuery(QueryConsts.SELECT_USERS_PROJECTS);
+        query.setParameter("userId", userId);
+        List<Project> ret = query.getResultList();
+        LOGGER.debug("getUsersProjects called with userId={}", userId);
+        return ret;
+    }
+
+    @Override
+    public boolean setProjectsTeam(List<User> addedUsers, BigInteger id) {
```

Comment:
The method is composition of dao method:
`boolean addUsersToProjects(UsersToProjects toAdd) {
...
}`
so in service layer it should contain interation through loop and invoke dao layer method with signature presented above